### DISCUSSION TITLE
Backup  of the existing .cfg files, or..

### DISCUSSION THREAD

**Discussion by Manfredo1952** [bassamanator/Sovol-SV06-firmware#90] [Q&A]
At the moment I am creating an .img file from the SD card within my BTT Pad 7, using ApplePi Baker - takes sbotu an hour to finish.  
Is there are faster way of creating a backup regularly ?
Thanks

**Comment by bassamanator** [comment]
What are you trying to backup specifially? The printer configuration?

**Comment by Manfredo1952** [comment]
Basically the files which are required to run the whole show. I would like to have a backup of a startup-SD Card
and then only have copy those files onto it, just in case the system fails after am update.

**Comment by bassamanator** [comment]
Well there are many approaches to this. You have to choose the one that works for you.

For me, I see little value in backing up the entire image (I used to do this myself btw). But then again, it's quite easy *for me* to push out a fresh new image to any one of my devices. I'm only interesting is saving things that aren't readily downloadable from the internet. The files that I would backup are the ones that are unique, and I would use something like `rsync` to do daily backups/weekly backups.
`BackupDeviceX.sh`
```sh
#!/bin/bash
sourceLocation=pi@ip.address.of.pi:
dest=/path/to/whereYouWantThingsSaved
exclude=**/node_modules
rsync "-avXP" "--info=progress2" "--no-i-r" "-R" "--exclude=${exclude}" "${sourceLocation}/home/pi/path/to/whatYouWantSaved" "${sourceLocation}/home/pi/path/to/whatYouWantSaved2" "${dest}"
```
Then just run via `./BackupDeviceX.sh`. You can run it over and over again, and only changed files will be `synced`.

Having said all this, you have to choose the method that you can reliably perform time and time again.
